python/src/equistore/operations/pow.py

```diff
@@ -45,16 +43,25 @@ def _pow_block_constant(block: TensorBlock, constant: float) -> TensorBlock:
         components=block.components,
         properties=block.properties,
     )
+    if len(block.gradients_list()) == 0:
+        return result_block
+
+    _shape = ()
+    for c in block.components:
+        _shape += (len(c),)
+    _shape += (len(block.properties),)
 
     for parameter, gradient in block.gradients():
         values_grad = []
-        for isample in range(len(block.samples)):
-            isample_grad1 = np.where(gradient.samples["sample"] == isample)[0]
-            values_grad.append(
-                constant
-                * gradient.data[isample_grad1]
-                * block.values[isample] ** (constant - 1)
+        diff_components = len(gradient.components) - len(block.components)
+        values_grad.append(
+            constant
+            * gradient.data[:]
```

Comment:
this should work as well

```suggestion
            * gradient.data
```